jerry-math/log2.c

```diff
@@ -52,6 +51,7 @@
 double
 log2 (double x)
 {
+  const double zero = 0.0;
```

Comment:
By doing this, can remove dive by zero warning/error on msvc

Reply:
Please describe what kind of diagnostics are reported by MSVC. Division by a zero double literal should behave just like a division by a constant zero double variable. Usually. In a decent compiler where constants are propagated.

Reply:
error messages：

```
build] [18/385   0% :: 0.065] Building C object jerry-math\CMakeFiles\jerry-math.dir\log10.c.obj
[build] FAILED: jerry-math/CMakeFiles/jerry-math.dir/log10.c.obj 
[build] C:\PROGRA~2\MICROS~1.0\VC\bin\amd64\cl.exe  /nologo  -I..\..\jerry-math\include /DWIN32 /D_WINDOWS /W3 /wd4996 /MD /O1 /Ob1 /DNDEBUG /showIncludes /Fojerry-math\CMakeFiles\jerry-math.dir\log10.c.obj /Fdjerry-math\CMakeFiles\jerry-math.dir\jerry-math.pdb /FS -c ..\..\jerry-math\log10.c
[build] ..\..\jerry-math\log10.c(87): error C2124: divide or mod by zero
[build] [18/385   0% :: 0.065] Building C object jerry-math\CMakeFiles\jerry-math.dir\log1p.c.obj
[build] FAILED: jerry-math/CMakeFiles/jerry-math.dir/log1p.c.obj 
[build] C:\PROGRA~2\MICROS~1.0\VC\bin\amd64\cl.exe  /nologo  -I..\..\jerry-math\include /DWIN32 /D_WINDOWS /W3 /wd4996 /MD /O1 /Ob1 /DNDEBUG /showIncludes /Fojerry-math\CMakeFiles\jerry-math.dir\log1p.c.obj /Fdjerry-math\CMakeFiles\jerry-math.dir\jerry-math.pdb /FS -c ..\..\jerry-math\log1p.c
[build] ..\..\jerry-math\log1p.c(125): error C2124: divide or mod by zero
[build] [18/385   0% :: 0.092] Building C object jerry-math\CMakeFiles\jerry-math.dir\asinh.c.obj
[build] [18/385   1% :: 0.093] Building C object jerry-math\CMakeFiles\jerry-math.dir\atan2.c.obj
[build] ..\..\jerry-math\atan2.c(77): warning C4146: unary minus operator applied to unsigned type, result still unsigned
[build] [18/385   1% :: 0.102] Building C object jerry-math\CMakeFiles\jerry-math.dir\asin.c.obj
[build] [18/385   1% :: 0.103] Building C object jerry-math\CMakeFiles\jerry-math.dir\acosh.c.obj
[build] [18/385   1% :: 0.106] Building C object jerry-math\CMakeFiles\jerry-math.dir\cbrt.c.obj
[build] [18/385   2% :: 0.110] Building C object jerry-math\CMakeFiles\jerry-math.dir\copysign.c.obj
[build] [18/385   2% :: 0.113] Building C object jerry-math\CMakeFiles\jerry-math.dir\ceil.c.obj
[build] ..\..\jerry-math\ceil.c(116): warning C4018: '<': signed/unsigned mismatch
[build] [18/385   2% :: 0.115] Building C object jerry-math\CMakeFiles\jerry-math.dir\floor.c.obj
[build] ..\..\jerry-math\floor.c(115): warning C4018: '<': signed/unsigned mismatch
[build] [18/385   2% :: 0.116] Building C object jerry-math\CMakeFiles\jerry-math.dir\atanh.c.obj
[build] c:\work\study\languages\typescript\jerryscript\jerry-math\atanh.c(69) : warning C4723: potential divide by 0
[build] [18/385   3% :: 0.116] Building C object jerry-math\CMakeFiles\jerry-math.dir\fabs.c.obj
[build] [18/385   3% :: 0.116] Building C object jerry-math\CMakeFiles\jerry-math.dir\cosh.c.obj
[build] [18/385   3% :: 0.118] Building C object jerry-math\CMakeFiles\jerry-math.dir\acos.c.obj
[build] [18/385   3% :: 0.119] Building C object jerry-math\CMakeFiles\jerry-math.dir\expm1.c.obj
[build] [18/385   4% :: 0.120] Building C object jerry-math\CMakeFiles\jerry-math.dir\atan.c.obj
[build] [18/385   4% :: 0.126] Building C object jerry-math\CMakeFiles\jerry-math.dir\fmod.c.obj
[build] ..\..\jerry-math\fmod.c(54): warning C4146: unary minus operator applied to unsigned type, result still unsigned
[build] [18/385   4% :: 0.126] Building C object jerry-math\CMakeFiles\jerry-math.dir\exp.c.obj
[build] ninja: build stopped: subcommand failed.
[build] Build finished with exit code 1

```

Reply:
on msvc, they becave differently, warning vs error